File: doc-builder/server/DocBuilder.Data/Entities/DocItem.cs
namespace DocBuilder.Data.Entities
{
    public class DocItem
    {
        public int Id { get; set; }
        public int DocId { get; set; }
        public DocItemType Type { get; set; }
        public int Index { get; set; }
        public string Value { get; set; }
        public bool AllowMultiple { get; set; }
        public bool IsDropdown { get; set; }

        public DocAnswer? Answer { get; set; }
        public Doc? Doc { get; set; }

        public ICollection<DocOption> Options { get; set; } = new List<DocOption>();

        public DocItem(string value, DocItemType type)
        {
            Value = value;
            Type = type;
        }
    }
}

File: doc-builder/src/core/src/lib/models/doc-item.ts
import { Doc } from './doc';
import { DocAnswer } from './doc-answer';
import { DocItemType } from './enums';
import { DocOption } from './doc-option';

export interface DocItem {
  id: number;
  docId: number;
  type: DocItemType;
  index: number;
  value: string;
  allowMultiple: boolean;
  isDropdown: boolean;

  answer?: DocAnswer;
  doc?: Doc;

  options: DocOption[];
}
